libraries/aws_ec2_security_groups.rb

```diff
@@ -15,63 +15,42 @@ class AwsEc2SecurityGroups < Inspec.resource(1)
     end
 EOX
 
-  # Constructor.  Args are reserved for row fetch filtering.
-  def initialize(raw_criteria = {})
-    validated_criteria = validate_filter_criteria(raw_criteria)
-    fetch_from_backend(validated_criteria)
-  end
+  include AwsPluralResourceMixin
 
   # Underlying FilterTable implementation.
   filter = FilterTable.create
   filter.add_accessor(:where)
         .add_accessor(:entries)
         .add(:exists?) { |x| !x.entries.empty? }
         .add(:group_ids, field: :group_id)
-  filter.connect(self, :access_key_data)
-
-  def access_key_data
-    @table
-  end
+  filter.connect(self, :table)
 
   def to_s
     'EC2 Security Groups'
   end
 
   private
 
-  def validate_filter_criteria(raw_criteria)
+  def validate_params(raw_criteria)
     unless raw_criteria.is_a? Hash
       raise 'Unrecognized criteria for fetching Security Groups. ' \
             "Use 'criteria: value' format."
     end
 
     # No criteria yet
-    recognized_criteria = check_criteria_names(raw_criteria)
-
-    recognized_criteria
-  end
-
-  def check_criteria_names(raw_criteria: {}, allowed_criteria: [])
-    # Remove all expected criteria from the raw criteria hash
-    recognized_criteria = {}
-    allowed_criteria.each do |expected_criterion|
-      recognized_criteria[expected_criterion] = raw_criteria.delete(expected_criterion) if raw_criteria.key?(expected_criterion)
-    end
-
-    # Any leftovers are unwelcome
     unless raw_criteria.empty?
-      raise ArgumentError, "Unrecognized filter criterion '#{raw_criteria.keys.first}'. Expected criteria: #{allowed_criteria.join(', ')}"
+      raise ArgumentError, 'aws_ec2_security_groups does not currently accept resource parameters.'
     end
-    recognized_criteria
+    raw_criteria
   end
 
-  def fetch_from_backend(criteria)
+  def fetch_from_api
     @table = []
-    backend = AwsEc2SecurityGroups::BackendFactory.create
+    backend = AwsEc2SecurityGroups::BackendFactory.create(inspec_runner)
     # Note: should we ever implement server-side filtering
     # (and this is a very good resource for that),
     # we will need to reformat the criteria we are sending to AWS.
-    backend.describe_security_groups(criteria).security_groups.each do |sg_info|
+    backend.describe_security_groups({}).security_groups.each do |sg_info|
```

Comment:
Why are we removing the criteria option here? Is there an issue with the new connection? or stability issue?

Reply:
Two reasons: 
1.  Using AwsPluralResourceMixin, we'd get our "criteria" / resource parameters from instance variables, not from the method argument.
2.  This method doesn't accept any resource params at this point (see line 42) and will validate for that.  So, it sends an empty hash.